server/test/forms/translation/ProgramTranslationFormTest.java

```diff
@@ -0,0 +1,195 @@
+          .put(ProgramTranslationForm.statusTextFieldName(0), "first status text")
+          .put(ProgramTranslationForm.statusEmailFieldName(0), "first status email")
+          .put(ProgramTranslationForm.configuredStatusFieldName(1), "second status configured text")
+          .put(ProgramTranslationForm.statusTextFieldName(1), "second status text")
+          .put(ProgramTranslationForm.statusEmailFieldName(1), "second status email")
+          .build();
+
+  @Test
+  public void bindFromRequest() throws Exception {
+    Request request = fakeRequest().bodyForm(REQUEST_DATA_WITH_TWO_TRANSLATIONS).build();
+
+    ProgramTranslationForm form =
+        ProgramTranslationForm.bindFromRequest(request, instanceOf(FormFactory.class), 2);
+    assertThat(form.getUpdateData())
+        .isEqualTo(
+            LocalizationUpdate.builder()
+                .setLocalizedDisplayName("display name")
+                .setLocalizedDisplayDescription("display description")
+                .setStatuses(
+                    ImmutableList.of(
+                        LocalizationUpdate.StatusUpdate.builder()
+                            .setConfiguredStatusText("first status configured text")
+                            .setLocalizedStatusText(Optional.of("first status text"))
+                            .setLocalizedEmailBody(Optional.of("first status email"))
+                            .build(),
+                        LocalizationUpdate.StatusUpdate.builder()
+                            .setConfiguredStatusText("second status configured text")
+                            .setLocalizedStatusText(Optional.of("second status text"))
+                            .setLocalizedEmailBody(Optional.of("second status email"))
+                            .build()))
+                .build());
+  }
+
+  @Test
+  public void bindFromRequest_extraStatusesInFormBodyBeyondSpecifiedAreIgnored() throws Exception {
+    Request request = fakeRequest().bodyForm(REQUEST_DATA_WITH_TWO_TRANSLATIONS).build();
+
+    ProgramTranslationForm form =
+        ProgramTranslationForm.bindFromRequest(request, instanceOf(FormFactory.class), 1);
+    assertThat(form.getUpdateData())
+        .isEqualTo(
+            LocalizationUpdate.builder()
+                .setLocalizedDisplayName("display name")
+                .setLocalizedDisplayDescription("display description")
+                .setStatuses(
+                    ImmutableList.of(
+                        LocalizationUpdate.StatusUpdate.builder()
+                            .setConfiguredStatusText("first status configured text")
+                            .setLocalizedStatusText(Optional.of("first status text"))
+                            .setLocalizedEmailBody(Optional.of("first status email"))
+                            .build()))
+                .build());
+  }
+
+  @Test
+  public void bindFromRequest_notEnoughStatusesInFormBodyAreOmmitted() throws Exception {
```

Comment:
I'm not really following what this test is testing.  The form body has 3 items (which indicates the DB had 3 correct?), and the request has 2, so the 3rd is dropped?

Reply:
This form is only responsible for parsing the form body data and doesn't have any true idea about what's configured inside the database outside of the argument passed to `bindFromRequest`.

In this case, we're testing an out-of-date-tab situation where the configured statuses count is 3 and there are only 2 provided in the form body. The goal is to ensure that trying to parse a non-present 3rd status wouldn't crash anything and would just return 2 at the end. Upstream callers would be responsible for caring about whether having 2 updates was acceptable or not (that's now documented in the implementation).

I updated the comment.